src/main/java/io/vlingo/http/resource/StaticFilesResource.java

```diff
@@ -54,53 +46,47 @@ public void serveFile(final String contentFile, final String root, final String
 
     final String uri = contentFile.isEmpty() ? "/index.html" : context.request.uri.toString();
 
-    try {
-      final String contentPath = String.join(" ", contentFilePath(rootPath + uri).split("%20"));
-      // logger().debug("contentPath: "+contentPath);
-      final byte[] fileContent = readFile(contentPath);
-      completes().with(Response.of(Ok,
-          Header.Headers.of(ResponseHeader.of(RequestHeader.ContentType, guessContentType(contentPath)),
-              ResponseHeader.of(ContentLength, fileContent.length)),
-          Body.from(fileContent, Body.Encoding.UTF8).content()));
-    } catch (IOException e) {
-      internalServerError(e);
-    } catch (IllegalArgumentException e) {
-      completes().with(Response.of(NotFound));
-    }
+    Response response = Arrays.asList(
+      rootPath + uri,
+      withIndexHtmlAppended(rootPath + uri)
+    ).stream()
+      .map(this::cleanPath)
+      .filter(this::isFile)
```

Comment:
`filter()` is a lazy operation. Since we use `findFirst()` next, filter's predicate won't be called on all elements of the collection, but only until the first file is found.

Reply:
@jakzal This looks promising. Do you plan to run with this? If you need tests on Windows and Linux @danilo-ambrosio and do both, and @hurelhuyag can verify Linux.

Reply:
It's still in draft, as I realized 404 pages are blank (i.e. `/`). I need to confirm whether this was always the case.